cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -402,6 +413,7 @@ private InetSocketAddress findLeader(String brokers, String topic, int partition
       } catch (Exception e) {
         LOG.error("Failed to communicate with broker {}:{} for leader lookup for topic-partition {}-{}",
                   broker.getKey(), broker.getValue(), topic, partition, e);
+        Throwables.propagate(e);
```

Comment:
When I checked the cluster where https://issues.cask.co/browse/CDAP-3798 was observed, I noticed that all the log messages were from this logging. Without propagating this exception, we might not able to detect it and sleep.